src/Version.h

```diff
@@ -17,26 +19,86 @@
+enum class CompareMode { AtLeast, LessThan };
 
-static_assert(sizeof(exact_width_mumble_raw_version_t) <= sizeof(mumble_raw_version_t),
-			  "Invalid scalar type used to represent Mumble versions");
+//
+// The mumble version format (v2) is a uint64:
+// reserved/unused  major   minor   patch
+// 0xFFFF           0xFFFF  0xFFFF  0xFFFF
+// (big-endian)
+//
+
+using mumble_raw_version_t = std::uint64_t;
+using component_t          = std::uint16_t;
+
+constexpr mumble_raw_version_t OFFSET_UNUSED = 48;
+constexpr mumble_raw_version_t OFFSET_MAJOR  = 32;
+constexpr mumble_raw_version_t OFFSET_MINOR  = 16;
+constexpr mumble_raw_version_t OFFSET_PATCH  = 0;
+constexpr mumble_raw_version_t FIELD_MASK    = 0xFFFF;
+constexpr mumble_raw_version_t FIELD_UNUSED  = (FIELD_MASK << Version::OFFSET_UNUSED);
+constexpr mumble_raw_version_t FIELD_MAJOR   = (FIELD_MASK << Version::OFFSET_MAJOR);
+constexpr mumble_raw_version_t FIELD_MINOR   = (FIELD_MASK << Version::OFFSET_MINOR);
+constexpr mumble_raw_version_t FIELD_PATCH   = (FIELD_MASK << Version::OFFSET_PATCH);
 
 constexpr mumble_raw_version_t UNKNOWN = 0;
 
-unsigned int getRaw(const QString &version = QLatin1String(MUMTEXT(MUMBLE_VERSION)));
+QString getRelease();
+mumble_raw_version_t getRaw(const QString &version = QLatin1String(MUMTEXT(MUMBLE_VERSION)));
+mumble_raw_version_t fromConfig(const QVariant &config);
 QString toString(mumble_raw_version_t version);
-bool get(int *major, int *minor, int *patch, const QString &version = QLatin1String(MUMTEXT(MUMBLE_VERSION)));
+QString toConfigString(mumble_raw_version_t version);
+bool get(Version::component_t *major, Version::component_t *minor, Version::component_t *patch, const QString &version = QLatin1String(MUMTEXT(MUMBLE_VERSION)));
+
+constexpr Version::component_t getMajor(Version::mumble_raw_version_t version) {
+	return static_cast< Version::component_t >((version & Version::FIELD_MAJOR) >> Version::OFFSET_MAJOR);
+}
+
+constexpr Version::component_t getMinor(Version::mumble_raw_version_t version) {
+	return static_cast< Version::component_t >((version & Version::FIELD_MINOR) >> Version::OFFSET_MINOR);
+}
+
+constexpr Version::component_t getPatch(Version::mumble_raw_version_t version) {
+	return static_cast< Version::component_t >((version & Version::FIELD_PATCH) >> Version::OFFSET_PATCH);
+}
+
+constexpr Version::mumble_raw_version_t toRaw(Version::component_t major, Version::component_t minor, Version::component_t patch) {
+	return ((static_cast< mumble_raw_version_t >(major) << Version::OFFSET_MAJOR)
+			| (static_cast< mumble_raw_version_t >(minor) << Version::OFFSET_MINOR)
+			| (static_cast< mumble_raw_version_t >(patch) << Version::OFFSET_PATCH));
+}
+
+constexpr void fromRaw(Version::mumble_raw_version_t version, Version::component_t *major, Version::component_t *minor, Version::component_t *patch) {
+	*major = Version::getMajor(version);
+	*minor = Version::getMinor(version);
+	*patch = Version::getPatch(version);
+}
```

Comment:
```suggestion
constexpr void fromRaw(Version::mumble_raw_version_t version, Version::component_t &major, Version::component_t &minor, Version::component_t &patch) {
	major = Version::getMajor(version);
	minor = Version::getMinor(version);
	patch = Version::getPatch(version);
}
```
If `nullptr` is not a valid argument value, references should be used instead of pointers